feature/google/src/test/java/com/revenuecat/purchases/google/StoreProductTest.kt

```diff
@@ -222,4 +222,85 @@ class StoreProductTest {
+            basePlanId = basePlanId,
+            offerId = null,
+            pricingPhases = listOf(PricingPhase(
+                billingPeriod = Period.create("P1M"),
+                recurrenceMode = RecurrenceMode.INFINITE_RECURRING,
+                billingCycleCount = 0,
+                price = Price(
+                    formatted = "",
+                    amountMicros = 0L,
+                    currencyCode = "",
+                )
+            )),
+            tags = emptyList(),
+            productDetails,
+            offerToken
+        )
+
+        val subscriptionOption2 = GoogleSubscriptionOption(
+            productId = productId,
+            basePlanId = basePlanId,
+            offerId = null,
+            pricingPhases = listOf(PricingPhase(
+                billingPeriod = Period.create("P1M"),
+                recurrenceMode = RecurrenceMode.INFINITE_RECURRING,
+                billingCycleCount = 0,
+                price = Price(
+                    formatted = "",
+                    amountMicros = 0L,
+                    currencyCode = "",
+                )
+            )),
+            tags = emptyList(),
+            productDetails,
+            offerToken
+        )
+
+        val storeProduct = GoogleStoreProduct(
+            productId = "product_id",
+            basePlanId = "base_plan_id",
+            type = ProductType.SUBS,
+            price = price,
+            title = "TITLE",
+            description = "DESCRIPTION",
+            period = Period.create("P1M"),
+            subscriptionOptions = SubscriptionOptions(listOf(subscriptionOption1, subscriptionOption2)),
+            defaultOption = subscriptionOption1,
+            productDetails = productDetails,
+            presentedOfferingIdentifier = "originalOfferingId"
+        )
+
+        val expectedOfferingId = "expectedOfferingId"
+        val copiedStoreProduct = storeProduct.copyWithOfferingId(expectedOfferingId) as GoogleStoreProduct
+
+        assertThat(copiedStoreProduct.id).isEqualTo(storeProduct.id)
+        assertThat(copiedStoreProduct.productId).isEqualTo(storeProduct.productId)
+        assertThat(copiedStoreProduct.basePlanId).isEqualTo(storeProduct.basePlanId)
+        assertThat(copiedStoreProduct.type).isEqualTo(storeProduct.type)
+        assertThat(copiedStoreProduct.price).isEqualTo(storeProduct.price)
+        assertThat(copiedStoreProduct.title).isEqualTo(storeProduct.title)
+        assertThat(copiedStoreProduct.description).isEqualTo(storeProduct.description)
+        assertThat(copiedStoreProduct.period).isEqualTo(storeProduct.period)
+        assertThat(copiedStoreProduct.productDetails).isEqualTo(storeProduct.productDetails)
+
+        assertThat(copiedStoreProduct.presentedOfferingIdentifier).isEqualTo(expectedOfferingId)
+        assertThat(copiedStoreProduct.defaultOption!!.presentedOfferingIdentifier).isEqualTo(expectedOfferingId)
+        copiedStoreProduct.subscriptionOptions?.forEach {
```

Comment:
Do we want to force unwrap the `subscriptionOptions` here to make sure these get tested?

```suggestion
        copiedStoreProduct.subscriptionOptions!!.forEach {
```

Reply:
@beylmk I think this still might be a good thing to add here? 🤔

Reply:
oh! i somehow missed this comment a couple days ago, sorry! yes i'll definitely add that